circuit/types/group/src/lib.rs

```diff
@@ -57,12 +57,43 @@ impl<E: Environment> Inject for Group<E> {
     /// For safety, the resulting point is always enforced to be on the curve with constraints.
     /// regardless of whether the y-coordinate was recovered.
     fn new(mode: Mode, group: Self::Primitive) -> Self {
-        // Initialize the x- and y-coordinate field elements.
-        let (x, y) = group.to_xy_coordinate();
-        let x = Field::new(mode, x);
-        let y = Field::new(mode, y);
+        // Inject the point.
+        let point = {
+            // Initialize the (x, y) coordinates of the point as field elements.
+            let (x, y) = group.to_xy_coordinate();
+            // Inject the `(x, y)` coordinates as field elements.
+            Self { x: Field::new(mode, x), y: Field::new(mode, y) }
+        };
+
+        // Inject `(point / COFACTOR)` as a witness.
+        let point_inv: Self = witness!(|point| point.div_by_cofactor());
```

Comment:
`witness!` does the following:
1. It ejects the `mode` of `point`, to determine whether to allocate `point_inv` as `Mode::Constant` or `Mode::Private`
2. It computes `point_inv` as `point.div_by_cofactor()` natively (via `console`)
3. It allocates `point_inv` as `circuit::Group::new(mode, point_inv)`

Reply:
Doesn't the call to `circuit::Group::new(mode, point_inv)` lead to an infinite loop?

Reply:
nvm, I see that you replaced it with individually allocations of the coordinates.

Reply:
Yeah, it was my bad there :)